### PR TITLE
Control event fix

### PR DESCRIPTION
**Pull request by vinay-ThoughtWorks**

### CODE REVIEW

File: src/components/Container.jsx [hunk 1]
```diff
@@ -30,9 +30,30 @@ export class Container extends addMoreDecorator(Component) {
   componentWillMount() {
     const initScript = this.props.metadata.events && this.props.metadata.events.onFormInit;
     if (initScript) {
-      const updatedTree = new ScriptRunner(this.state.data, this.props.patient).execute(initScript);
-      this.setState({ data: updatedTree });
+      const ut = new ScriptRunner(this.state.data, this.props.patient).execute(initScript);
+      this.setState({ data: ut });
```
Comment (shivarach) [inline src/components/Container.jsx]: Looks like only variable rename? `updatedTree` is better.

File: src/components/ObsControl.jsx [hunk 2]
```diff
@@ -25,13 +25,17 @@ export class ObsControl extends addMoreDecorator(Component) {
     this.setAbnormal = this.setAbnormal.bind(this);
   }
 
-  onValueChangeDone() {
-    if (this.props.onEventTrigger) {
+  isAllowedToTriggerControlEvent(triggerControlEvent) {
+    return triggerControlEvent === undefined || triggerControlEvent;
```
Comment (shivarach) [inline src/components/ObsControl.jsx:29]: It says `allowToTriggerControlEvent` when `triggerControlEvent` is `undefined`. When it is `undefined`?

**Review by shivarach — CHANGES_REQUESTED** [review]

File: src/components/ObsControl.jsx (hunk 2 above)
Comment (vinay-ThoughtWorks) [reply src/components/ObsControl.jsx:29]: Other than `componentDidMount`, every where it is undefined. That means after component mounted successfully and when some change is performed on that component(then we trigger the `onChange` inherited from the parent), we don't pass triggerControlEvent). It then takes as `undefined`.

**Comment by shivarach** [comment]
LGTM

**Pull request merged**